ruby.rb

```diff
@@ -3,49 +3,51 @@ class ArticlesController < ApplicationController
 
   def index
     @articles = Article.all.includes(:user)
+    articles_type
+    boundary = articles_boundary(0, 20)
+    offset = boundary[offset]
+    limit = boundary[limit]
+    @articles = @articles.order(created_at: :desc).offset(offset).limit(limit)
```

Comment:
After the entire refactoring is done, this should be a one liner. Read my comments below to understand this